pkg/cmd/base/options.go

```diff
@@ -20,3 +29,48 @@ func NewOptions(ioStreams util.IOStreams) *Options {
 		IOStreams: ioStreams,
 	}
 }
+
+// AddOutputFlags adds flags to adjust the output to a cobra command
+func (o *Options) AddOutputFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "One of 'yaml' or 'json'.")
+}
+
+// PrintObject prints an object to IOStreams.out, using o.Output to print in the selected output format
+func (o *Options) PrintObject(obj interface{}) error {
+	switch o.Output {
+	case "":
+		fmt.Fprintf(o.IOStreams.Out, "%v", obj)
+
+	case "yaml":
+		marshalled, err := yaml.Marshal(&obj)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintln(o.IOStreams.Out, string(marshalled))
+
+	case "json":
+		marshalled, err := json.MarshalIndent(&obj, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintln(o.IOStreams.Out, string(marshalled))
+
+	default:
+		// There is a bug in the program if we hit this case.
+		// However, we follow a policy of never panicking.
```

Comment:
I would prefer a policy of making assertions or panic if the assertions are not fulfilled.

Reply:
It‘s easier to test if we return an error instead